src/resnet50.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{model::*, input::*};
+use egg::*;
+
+
+fn resnet_block(graph: &mut GraphConverter, mut input: Id, strides: (i32, i32), out_channels: i32, input_dim_1: i32, name_gen: &mut NameGen) -> Id {
+    let w1 = graph.new_input(&name_gen.get_name(), out_channels, input_dim_1, 1, 1);
+    let t = graph.conv2d(input, w1, 1, 1, PSAME, ACTRELU);
+    let w2 = graph.new_input(&name_gen.get_name(), out_channels, out_channels, 3, 3);
+    let t = graph.conv2d(t, w2, strides.0, strides.1, PSAME, ACTRELU);
+    let w3 = graph.new_input(&name_gen.get_name(), out_channels*4, out_channels, 1, 1);
```

Comment:
It seems that the `graph.new_input(&name_gen.get_name(), ...)` pattern comes up often. You might want to fold `NameGen` into `GraphConverter` and have a dedicated method to add an input with a fresh name. This would simplify calls to `new_input`.

Reply:
Moved NameGen into GraphConverter